prompting/rewards/streaming.py

```diff
@@ -0,0 +1,48 @@
+import time
+import torch
+from prompting.dendrite import DendriteResponseEvent
+from prompting.rewards import (
+    BaseRewardModel,
+    BatchRewardOutput,
+)
+
+class StreamingRewardModel(BaseRewardModel):
+    @property
+    def name(self) -> str:
+        return "streaming"
+
+    def __init__(self, max_tokens_per_chunk:int, **kwargs):
+        super().__init__()
+        self.max_tokens_per_chunk = max_tokens_per_chunk
+
+
+    def reward(self, _: str, response_event: DendriteResponseEvent) -> BatchRewardOutput:
+        """Compute difference scores given a completion and reference pair."""       
+        rewards = []
+        timings = []
+        penalty_per_exceeding_chunk = 0.25
+                
+        # Iterate through each chunk of response tokens
+        for response_tokens_per_chunks in response_event.stream_results_all_tokens_per_chunk:            
+            start_time = time.time()
+             
+            # Calculate the accumulated penalty for the current chunk
+            accumulated_penalty = sum(
+                penalty_per_exceeding_chunk if tokens_per_chunk > self.max_tokens_per_chunk else 0 
+                for tokens_per_chunk in response_tokens_per_chunks
+            )
+            
+            # Record the timing for this computation
+            timings.append(time.time() - start_time)
```

Comment:
Minor suggestion is to measure with `perf_counter` which is more precise and not affected by system clock

```suggestion
            start_time = time.perf_counter()
             
            # Calculate the accumulated penalty for the current chunk
            accumulated_penalty = sum(
                penalty_per_exceeding_chunk if tokens_per_chunk > self.max_tokens_per_chunk else 0 
                for tokens_per_chunk in response_tokens_per_chunks
            )
            
            # Record the timing for this computation
            timings.append(time.perf_counter() - start_time)
```